internal/applicationsnapshot/input_test.go

```diff
@@ -0,0 +1,70 @@
+package applicationsnapshot
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	appstudioshared "github.com/redhat-appstudio/managed-gitops/appstudio-shared/apis/appstudio.redhat.com/v1alpha1"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_DetermineInputSpec(t *testing.T) {
+	imageRef := "quay.io://redhat-appstudio/ec"
+	snapshot := &appstudioshared.ApplicationSnapshotSpec{
+		Components: []appstudioshared.ApplicationSnapshotComponent{
+			{
+				Name:           "Unnamed",
+				ContainerImage: imageRef,
+			},
+		},
+	}
+	testJson, _ := json.Marshal(snapshot)
+	tests := []struct {
+		filePath string
+		input    string
+		imageRef string
+		want     *appstudioshared.ApplicationSnapshotSpec
+	}{
+		{
+			filePath: "/home/list-of-images.json",
+			input:    "",
+			imageRef: "",
+			want:     snapshot,
+		},
+		{
+			filePath: "",
+			input:    string(testJson),
+			imageRef: "",
+			want:     snapshot,
+		},
+		{
+			filePath: "",
+			input:    "",
+			imageRef: imageRef,
+			want:     snapshot,
+		},
+	}
+
+	savedReadfile := readFile
+	defer func() { readFile = savedReadfile }()
+
+	readFile = func(fs afero.Fs, filename string) ([]byte, error) {
+		return testJson, nil
+	}
+
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("DetermineInputSpec=%d", i), func(t *testing.T) {
+
+			got, err := DetermineInputSpec(tc.filePath, tc.input, tc.imageRef)
+			if err != nil {
+				t.Fatalf("error: %v", err)
+			} else if !assert.ObjectsAreEqual(tc.want, got) {
+				t.Fatalf("got %v; want %v", got, tc.want)
+			} else {
+				t.Logf("Success!")
+			}
```

Comment:
This gives us nicer messages on test failures and is also a bit terser
```suggestion
			assert.NoError(t, err)
			assert.Equal(t, tc.want, got)
```